doc/specs/stdlib_io.md

```diff
@@ -131,3 +131,130 @@ program demo_savetxt
     call savetxt('example.dat', x) 
 end program demo_savetxt
 ```
+
+## `disp` - display your data
+
+### Status
+
+Experimental
+
+### Class
+
+Impure subroutine.
+
+### Description
+
+Outputs a `logical/integer/real/complex/character/string_type` scalar or `logical/integer/real/complex` and rank-1/rank-2 array to the screen or a file `unit`.
+
+#### More details
+
+```fortran
+call disp( A(i, j, 2, :, 1:10) [, header, unit, brief] )    !! `i, j, ...` can be determined by `do` loop.
+```
+
+For `complex` type, the output format is `*(A25, 1X)`; 
+For    other types, the output format is `*(A12, 1X)`.
+
+To prevent users from accidentally passing large-length arrays to `disp`, causing unnecessary io blockage:
+1. If the `brief` argument is not specified, `disp` will print **the brief array content with a length of `10*50` by default**.
+2. Specify `brief=.true.`, `disp` will print **the brief array content with a length of `5*5`**;
+3. Specify `brief=.false.`, `disp` will print **`all` the contents of the array**.
+
+### Syntax
+
+`call [[stdlib_io(module):disp(interface)]]([x, header, unit, brief])`  
+
+### Arguments
+
+`x`: Shall be a `logical/integer/real/complex/string_type` scalar or `logical/integer/real/complex` and rank-1/rank-2 array.
```

Comment:
Why not array of characters or of string_type?

Reply:
The **character length** of array of `string_type` is often **uncertain**, and it is difficult to print them uniformly.
And we already have `stringlist`, its internal elements are **private**, and the outside is inaccessible.
(see [stdlib_stringlist_type.f90#L53~L54](https://github.com/fortran-lang/stdlib/blob/0d1cc1c171e670a004f8bd472ab4181434f2b734/src/stdlib_stringlist_type.f90#L53~L54))

Reply:
Support for `string_type` type is added: `string_type` scalar, rank-1/rank-2 array.
Only `character(*)` **scalar** is supported, `string_type` array should be more easy to use than `character(*)` array.
:)